src/main/scala/io/iohk/ethereum/vm/ProgramContext.scala

```diff
@@ -1,13 +1,51 @@
 package io.iohk.ethereum.vm
 
-import akka.util.ByteString
+import io.iohk.ethereum.domain._
 
+
+object ProgramContext {
+  def apply(stx: SignedTransaction, blockHeader: BlockHeader, world: WorldStateProxy): ProgramContext = {
+    import stx.tx
+
+    val senderAddress = stx.recoveredSenderAddress.get // FIXME: get, it should be validated but...
```

Comment:
> Yes, we need to be able to create a Transaction without a signature.

But why? Wasn't that meant for an _internal transaction_ which we're not going to use?